### DISCUSSION TITLE
multiple client_ids for each <Location>

### DISCUSSION THREAD

**Discussion by ymttto** [OpenIDC/mod_auth_openidc#798] [Q&A] [answered]
I want to specify multiple client_ids.

I want to divide one <VirtualHost> into multiple <Location> and specify a different client_id for each <Location>.
The OP itself is the same.

example:
https://hoge.example.com/aaa/xxxx => client_id: AAA
https://hoge.example.com/bbb/xxxx => client_id: BBB

The reasons for this are as follows.

There is only one login session (because there is also one OP) regardless of the path in https://hoge.example.com/ as a whole.
However, I want to change the login screen depending on the path (OP can change the appearance of the login screen by client_id).

Is this possible with the mod_auth_openidc setting?
(Can I set the OIDC Client ID in <Locaiton>?)

**Comment by zandbelt** [comment, marked as answer]
no, multiple client id's per Location is not possible, you'd need multiple vhosts for that; anyhow, typically your use case is solved by adding custom parameters to the authentication request (or using a different scope), see: https://github.com/zmartzone/mod_auth_openidc/wiki#13-how-can-i-add-custom-parameters-to-the-authorization-request

**Reply by ymttto** [reply]
Thank you for your answer. I understand.

Is it correct to understand that your answer "can be solved by custom parameters" is if we can change the behavior(*) of OP by custom parameters instead of client_id?

(*) the appearance of the login screen

**Reply by ymttto** [reply]
In other words, if I want to convey different information to the OP for each path, is it the only way to use custom parameters or different scope?

**Reply by zandbelt** [reply]
yes

**Reply by ymttto** [reply]
thank you. I understand.
